crucible/src/Lang/Crucible/Simulator/ExecutionTree.hs

```diff
@@ -1075,7 +1076,26 @@ ctxSymInterface = to (\ctx ->
 functionBindings :: Lens' (SimContext p sym ext) (FunctionBindings p sym ext)
 functionBindings = lens _functionBindings (\s v -> s { _functionBindings = v })
 
--- | Access the custom user-state inside the 'SimContext'.
+-- | Custom state inside the 'SimContext'.
+--
+-- Crucible itself is entirely polymorphic over @p@, downstream applications can
+-- instantiate it to any sort of state that they would like to associate with a
+-- 'SimContext'.
+--
+-- For example, Macaw-based applications can instantiate this to
+-- a structure holding enough information to perform incremental code discovery,
+-- see @AmbientSimulatorState@ in the @ambient-verifier@ project.
```

Comment:
Rather than linking to `ambient-verifier` (yet another project), how about we just link to [`MacawLazySimulatorState`](https://github.com/GaloisInc/macaw/blob/cbec559b428fdd194398d07fc08c8c570a1d3bab/symbolic/src/Data/Macaw/Symbolic/MemOps.hs#L368-L375) in `macaw-symbolic`? That would also segue more nicely with the reference to `HasMacawLazySimulatorState` below.

Reply:
Eh, if you don't mind I'd actually like to keep this reference. First of all, it offers a different example of things to put in the personality (incremental code discovery, rather than lazy memory). Secondly, I think it's nice to offer a variety of downstream projects as examples / inspiration.